src/dash/models/DashManifestModel.js

```diff
@@ -230,23 +231,43 @@ function DashManifestModel() {
     }
 
     function getViewpointForAdaptation(adaptation) {
-        return adaptation && adaptation.hasOwnProperty(DashConstants.VIEWPOINT) ? adaptation.Viewpoint : null;
+        if (!adaptation || !adaptation.hasOwnProperty(DashConstants.VIEWPOINT_ASARRAY) || !adaptation.Viewpoint_asArray.length) return [];
+        return adaptation.Viewpoint_asArray.map( viewpoint => {
```

Comment:
Suggest to use `VIEWPOINT_ASARRAY` here as well: `adaptation[DashConstants.VIEWPOINT_ASARRAY]`. Same for the other calls below